nixos/modules/services/databases/openldap.nix

```diff
@@ -137,25 +238,36 @@ in
         default = null;
         description = ''
           Password file for the root user.
-          The file should contain the string <literal>rootpw</literal> followed by the password.
-          e.g.: <literal>rootpw mysecurepassword</literal>
+
+          If the deprecated <literal>extraConfig</literal> or
+          <literal>extraDatabaseConfig</literal> options are set, this should
+          contain <literal>rootpw</literal> followed by the password
+          (e.g. <literal>rootpw thePasswordHere</literal>).
+
+          Otherwise the file should contain only the password (no trailing
+          newline or leading <literal>rootpw</literal>).
         '';
       };
 
       logLevel = mkOption {
-        type = types.str;
-        default = "0";
-        example = "acl trace";
-        description = "The log level selector of slapd.";
+        type = types.nullOr (types.listOf types.str);
+        default = null;
+        example = literalExample "[ \"acl\" \"trace\" ]";
```

Comment:
This looks backwards incompatible, but you could use `types.coercedTo types.str (<function to change "acl trace" to ["acl" "trace"]>) (types.listOf types.str)`

Reply:
Done, with `splitString`.